glean-core/src/core/mod.rs

```diff
@@ -711,14 +711,52 @@ impl Glean {
     /// # Arguments
     ///
     /// * `json` - The stringified JSON representation of a `MetricsEnabledConfig` object
-    pub fn set_metrics_enabled_config(&self, cfg: MetricsEnabledConfig) {
-        // Set the current MetricsEnabledConfig, keeping the lock until the epoch is
-        // updated to prevent against reading a "new" config but an "old" epoch
-        let mut lock = self.remote_settings_metrics_config.lock().unwrap();
-        *lock = cfg;
-
-        // Update remote_settings epoch
-        self.remote_settings_epoch.fetch_add(1, Ordering::SeqCst);
+    pub fn set_metrics_enabled_config(&self, feature_id: String, cfg: MetricsEnabledConfig) {
+        let mut remote_settings_metrics_config =
+            self.remote_settings_metrics_config.lock().unwrap();
+        if let Some(feature_metric_config) = remote_settings_metrics_config.get(&feature_id) {
+            // Set the current MetricsEnabledConfig, keeping the lock until the epoch is
+            // updated to prevent against reading a "new" config but an "old" epoch
+            let mut lock = feature_metric_config.config.lock().unwrap();
+            *lock = cfg;
+
+            // Update the epoch for this feature
+            feature_metric_config.epoch.fetch_add(1, Ordering::SeqCst);
+        } else {
+            // There isn't an existing feature-id so create a new entry for it with the
+            // configuration that was provided. We start with a value of 1 in order to allow
+            // metrics to catch the first update since they start with a 0.
+            remote_settings_metrics_config.insert(
+                feature_id,
+                FeatureMetricConfiguration {
+                    epoch: AtomicU8::new(1),
+                    config: Arc::new(Mutex::new(cfg)),
+                },
+            );
+        }
+    }
+
+    /// Performs a lookup by metric_id (<category>.<name>) to get the feature identifier
+    /// associated with it if the metric has a remote configuration applied. Used by
+    /// created metrics to check if there is an updated remote configuration to apply for
+    /// the associated feature.
+    ///
+    /// # Arguments
+    ///
+    /// * `metric_id` - The metric identifier in the format <category>.<name>
+    pub(crate) fn get_feature_id_for_metric(&self, metric_id: String) -> Option<String> {
```

Comment:
Yeah, nothing prevents that aside from careful experiment planning at the moment. This is a known limitation that will go away once Nimbus completes the multi-feature work that is currently in progress. The current risks are minimal because right now there is only one team (Search & Suggest) that's using this so it's been possible to coordinate the upcoming experiments.